lib/syskit/network_generation/system_network_generator.rb

```diff
@@ -343,6 +386,40 @@ def self.verify_device_allocation(plan, toplevel_tasks_to_requirements = {})
                 end
             end
 
+            def self.verify_all_deployments_are_unique(
+                plan,
+                default_deployment_group,
+                toplevel_tasks_to_requirements
+            )
+                deployment_to_task_map = {}
+                plan.find_local_tasks(Syskit::TaskContext).each do |t|
+                    deployment_to_task_map[t.orocos_name] =
+                        (deployment_to_task_map[t.orocos_name] || []) + [t]
+                end
+
+                using_same_deployment = deployment_to_task_map.select do |_, tasks|
+                    tasks.size > 1
+                end
+
+                return if using_same_deployment.empty?
+
+                deployment_to_task = using_same_deployment
+                                     .each_with_object({}) do |(orocos_name, tasks), h|
+                    deployed_tasks = default_deployment_group
+                                     .find_all_suitable_deployments_for(tasks.first)
+
+                    deployed_task = deployed_tasks.select do |d|
+                        d.mapped_task_name == orocos_name
+                    end
+
+                    h[deployed_task.first] = tasks
+                end
```

Comment:
The deployment is not guaranteed to be coming from the default deployment group.

But, in any case, the information you're looking for is already in Deployment. Specifically, `Deployment.model.deployed_orogen_model_by_name`. The type won't be the same, but it IMO has what you need.